src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStoreConfig.java

```diff
@@ -104,16 +110,14 @@ public String getDisplayName() {
 
     @Override
     public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
-        boolean ret = super.configure(req, json);
-        if (ret){
-            save();
-        }
-        return ret;
+        super.configure(req, json);
+        save();
```

Comment:
Delete this override and `save()` from within each setter.

Reply:
GlobalConfiguration does not use BulkChange, so the file it is saved 3 times, I just test it. I will save on the setters, now I understand the System Configuration page takes a lot to save.

Reply:
Yes any optimization here would be in core via `BulkChange`. You generally need save in setters anyway, for example to support JEP-201 I think, which would not be calling the web-oriented `configure`.

Reply:
roger that